### ISSUE TITLE
Add common network/DNS tools to debugging?

### ISSUE THREAD

**Issue by phillbaker** [missing-semester/missing-semester#166]
Thanks for the great content in this series.

In the "Debugging" lecture, it would be useful to add `netcat` or `telnet` to the list of tools to use when debugging network issues. As well, listing `dig` or `nslookup` and `host` or `getent hosts` would be helpful in cases of debugging DNS issues. 

A couple of examples where these are used from my experiences as a software engineer:
* working through issues with firewalls in public cloud or on premise
* debugging VPN setup issues

**Comment by jonhoo** [comment]
This is good advice for future years! cc @JJGO @anishathalye

**Comment by JJGO** [comment]
Agreed, thanks for the suggestions. If we get into networking, I'd definitely like to mention a few more tools:

- `tcpdump` - such a great tool
- `netstat` - aka "what process is using that port" tool
- `curl` / Postman - debugging API calls
- `ping` and `mtr`
- `nmap`
- `nethogs`

Actually, I think we could easily devote an entire lecture on networking. We could also include some fun stuff like mac spoofing.

**Comment by flindeberg** [comment]
Some more potentially interesting topics in the realm of networking:
- `dns` as in the protocol and actual implementation
- `delv`, i.e. a `dig`-like tool with some DNSSEC-magic
- Captive portals, (CG)NAT and similar solutions
- `.local` domain, `mDNS`, `zeroconf` and the interaction with Chromecasts, AppleTV, etc
